examples/llamacheck/llamacheck.cpp

```diff
@@ -0,0 +1,187 @@
+
+        fflush(stderr);
+
+        // create a llama_batch with size 512
+        // we use this object to submit token data for decoding
+
+        llama_batch batch = llama_batch_init(512, 0, 1);
+
+        // evaluate the initial prompt
+        for (size_t i = 0; i < tokens_list.size(); i++) {
+            llama_batch_add(batch, tokens_list[i], i, { 0 }, false);
+        }
+
+        // llama_decode will output logits only for the last token of the prompt
+        batch.logits[batch.n_tokens - 1] = true;
+
+        if (llama_decode(ctx, batch) != 0) {
+            LOG_TEE("%s: llama_decode() failed\n", __func__);
+            return 1;
+        }
+
+        int n_cur    = batch.n_tokens;
+        int n_decode = 0;
+
+        const auto t_main_start = ggml_time_us();
+
+        while (n_cur <= n_len) {
+            // sample the next token
+            {
+                auto   n_vocab = llama_n_vocab(model);
+                auto * logits  = llama_get_logits_ith(ctx, batch.n_tokens - 1);
+
+                std::vector<llama_token_data> candidates;
+                candidates.reserve(n_vocab);
+
+                for (llama_token token_id = 0; token_id < n_vocab; token_id++) {
+                    candidates.emplace_back(llama_token_data{ token_id, logits[token_id], 0.0f });
+                }
+
+                llama_token_data_array candidates_p = { candidates.data(), candidates.size(), false };
+
+                // sample the most likely token
+                const llama_token new_token_id = llama_sample_token_greedy(ctx, &candidates_p);
+
+                // is it an end of generation?
+                if (llama_token_is_eog(model, new_token_id) || n_cur >= n_len) {
+                    LOG_TEE("\n");
+
+                    break;
+                }
+
+                LOG_TEE("%s", llama_token_to_piece(ctx, new_token_id).c_str());
+                fflush(stdout);
+
+                // prepare the next batch
+                llama_batch_clear(batch);
+
+                // push this new token for next evaluation
+                llama_batch_add(batch, new_token_id, n_cur, { 0 }, true);
+
+                n_decode += 1;
+            }
+
+            n_cur += 1;
+
+            // evaluate the current batch with the transformer model
+            if (llama_decode(ctx, batch)) {
+                fprintf(stderr, "%s : failed to eval, return code %d\n", __func__, 1);
+                return 1;
+            }
+        }
+        LOG_TEE("\n");
+
+        const auto t_main_end = ggml_time_us();
+
+        LOG_TEE("%s: decoded %d tokens in %.2f s, speed: %.2f t/s\n",
+                __func__, n_decode, (t_main_end - t_main_start) / 1000000.0f, n_decode / ((t_main_end - t_main_start) / 1000000.0f));
+
+        llama_print_timings(ctx);
+
+        fprintf(stderr, "\n");
+
+        llama_batch_free(batch);
+        LOG_TEE(">>>\n");
+    }
+
+
+    llama_free(ctx);
+    llama_free_model(model);
+
+    llama_backend_free();
+
+    return 0;
+}
```

Comment:
Also needs a final newline at the end of the file.

```suggestion
}

```